prov/efa/src/efa_tostr.c

```diff
@@ -0,0 +1,109 @@
+		[FI_EFA_FLUSHED]                       = "Flushed during QP destroy",
+		[FI_EFA_LOCAL_ERROR_QP_INTERNAL_ERROR] = "Internal queue pair error",
+		[FI_EFA_LOCAL_ERROR_INVALID_OP_TYPE]   = "Invalid operation type",
+		[FI_EFA_LOCAL_ERROR_INVALID_AH]        = "Invalid address handle",
+		[FI_EFA_LOCAL_ERROR_INVALID_LKEY]      = "Invalid local key (LKEY)",
+		[FI_EFA_LOCAL_ERROR_BAD_LENGTH]        = "Message too long",
+		[FI_EFA_REMOTE_ERROR_BAD_ADDRESS]      = "Invalid address",
+		[FI_EFA_REMOTE_ERROR_ABORT]            = "Remote connection aborted",
+		[FI_EFA_REMOTE_ERROR_BAD_DEST_QPN]     = "Invalid destination queue pair number (QPN)",
+		[FI_EFA_REMOTE_ERROR_RNR]              = "Destination receiver not ready",
+		[FI_EFA_REMOTE_ERROR_BAD_LENGTH]       = "Remote SGL too short",
+		[FI_EFA_REMOTE_ERROR_BAD_STATUS]       = "Unexpected status received from remote",
+		[FI_EFA_LOCAL_ERROR_UNRESP_REMOTE]     = "Unresponsive remote",
+	};
+
+	return (status < FI_EFA_OK || status > FI_EFA_LOCAL_ERROR_UNRESP_REMOTE)
+		? UNKNOWN_ERR_STR
+		: status_str[status];
+}
+
+static const char *efa_errno_str(int err)
+{
+	static const char *errno_str[] = {
+		[FI_EFA_ERR_OTHER                   - FI_EFA_ERRNO_OFFSET] = UNKNOWN_ERR_STR,
+		[FI_EFA_ERR_DEPRECATED_PKT_TYPE     - FI_EFA_ERRNO_OFFSET] = "Deprecated packet type encountered",
+		[FI_EFA_ERR_INVALID_PKT_TYPE        - FI_EFA_ERRNO_OFFSET] = "Invalid packet type encountered",
+		[FI_EFA_ERR_UNKNOWN_PKT_TYPE        - FI_EFA_ERRNO_OFFSET] = "Unknown packet type encountered",
+		[FI_EFA_ERR_PKT_POST                - FI_EFA_ERRNO_OFFSET] = "Failure to post packet",
+		[FI_EFA_ERR_PKT_POST_OR_QUEUE       - FI_EFA_ERRNO_OFFSET] = "Failure to post or queue packet",
+		[FI_EFA_ERR_PKT_SEND_QUEUED         - FI_EFA_ERRNO_OFFSET] = "Encountered error sending queued packet",
+		[FI_EFA_ERR_PKT_PROC_MSGRTM         - FI_EFA_ERRNO_OFFSET] = "Error processing non-tagged RTM",
+		[FI_EFA_ERR_PKT_PROC_TAGRTM         - FI_EFA_ERRNO_OFFSET] = "Error processing tagged RTM",
+		[FI_EFA_ERR_PKT_ALREADY_PROCESSED   - FI_EFA_ERRNO_OFFSET] = "Packet already processed",
+		[FI_EFA_ERR_OOM                     - FI_EFA_ERRNO_OFFSET] = "Out of memory",
+		[FI_EFA_ERR_MR_CLOSE                - FI_EFA_ERRNO_OFFSET] = "MR close error",
+		[FI_EFA_ERR_MR_DEREG                - FI_EFA_ERRNO_OFFSET] = "MR deregistration error",
+		[FI_EFA_ERR_RX_ENTRY_COPY           - FI_EFA_ERRNO_OFFSET] = "RX entry copy error",
+		[FI_EFA_ERR_RX_ENTRIES_EXHAUSTED    - FI_EFA_ERRNO_OFFSET] = "RX entries exhausted",
+		[FI_EFA_ERR_TX_ENTRIES_EXHAUSTED    - FI_EFA_ERRNO_OFFSET] = "TX entries exhausted",
+		[FI_EFA_ERR_AV_INSERT               - FI_EFA_ERRNO_OFFSET] = "Failure inserting address into address vector",
+		[FI_EFA_ERR_RMA_ADDR                - FI_EFA_ERRNO_OFFSET] = "RMA address verification failed",
+		[FI_EFA_ERR_BULK_POST_RECV_BUF      - FI_EFA_ERRNO_OFFSET] = "Bulk post internal receive buffers",
```

Comment:
This error string doesn't indicate what the error was. Bulk posting internal receive buffers is not an error